bridge/Mixer.cpp

```diff
@@ -230,69 +260,39 @@ void Mixer::stopTransports()
 
         logTransportPacketLoss("", *videoStreamEntry.second->transport, _loggableId.c_str());
         videoStreamEntry.second->transport->stop();
-        if (!waitForPendingJobs(700, 5, *videoStreamEntry.second->transport))
-        {
-            logger::error("Transport for endpointId %s did not finish pending jobs in time. Continuing "
-                          "deletion anyway.",
-                _loggableId.c_str(),
-                videoStreamEntry.second->endpointId.c_str());
-        }
     }
 
     for (auto& barbell : _barbells)
     {
         logTransportPacketLoss(barbell.second->id, *barbell.second->transport, _loggableId.c_str());
         barbell.second->transport->stop();
-        if (!waitForPendingJobs(700, 5, *barbell.second->transport))
-        {
-            logger::error("Transport for barbell %s did not finish pending jobs in time. Continuing "
-                          "deletion anyway.",
-                _loggableId.c_str(),
-                barbell.second->id.c_str());
-        }
     }
 
     _barbellPorts.clear();
 }
 
-bool Mixer::waitForAllPendingJobs(const uint32_t timeoutMs)
+bool Mixer::hasPendingJobs() const
 {
-    uint32_t totalSleepTimeMs = 0;
-    const uint32_t waitSlice = 10;
-    bool pendingJobs = true;
-
-    while (totalSleepTimeMs < timeoutMs && pendingJobs)
+    static const auto hasPendingJobsPredicate = [] (auto& stream)
     {
-        pendingJobs = false;
-
-        for (auto& audioStream : _audioStreams)
-        {
-            if (audioStream.second->transport && audioStream.second->transport->hasPendingJobs())
-            {
-                pendingJobs = true;
-                break;
-            }
-        }
-
-        for (auto& videoStream : _videoStreams)
-        {
-            if (videoStream.second->transport && videoStream.second->transport->hasPendingJobs())
-            {
-                pendingJobs = true;
-                break;
-            }
-        }
+        return stream.second->transport && stream.second->transport->hasPendingJobs();
+    };
 
-        if (pendingJobs)
-        {
-            usleep(waitSlice * 1000);
-            totalSleepTimeMs += waitSlice;
-        }
+    static const auto recordingHasPendingJobsPredicate = [] (auto& recordingStream)
+    {
+            const auto& transports = recordingStream.second->_transports;
+            return std::find_if(transports.begin(), transports.end(), [] (auto& transport) { return transport.second->hasPendingJobs(); }) != transports.end();
+    };
 
-        _engineMixer.flush();
-    }
+    bool hasPendingJobs = _pendingJobsAsyncWaiter.hasTaskWaiting();
```

Comment:
Collapsing the whole implementatioin, we could have put all the shared_ptr<RtcTransports> in a vector and use a find_if like below but with one predicate and run this on a timerjob to check it and then set a flag or report to mixermanager when it is done.